src/lua/datetime.lua

```diff
@@ -915,6 +900,36 @@ local function datetime_parse_from(str, obj)
     end
 end
 
+local function datetime_cmp(lhs, rhs, is_raising)
+    if is_string(lhs) then
+        lhs = datetime_parse_full(lhs, nil, 0, true)
+    end
+    if is_string(rhs) then
+        rhs = datetime_parse_full(rhs, nil, 0, true)
+    end
+    if not is_datetime(lhs) or not is_datetime(rhs) then
+        if is_raising then
+            error('incompatible types for datetime comparison', 3)
+        else
+            return nil
+        end
+    end
+    local sdiff = lhs.epoch - rhs.epoch
+    return sdiff ~= 0 and sdiff or (lhs.nsec - rhs.nsec)
+end
+
+local function datetime_eq(lhs, rhs)
+    return datetime_cmp(lhs, rhs, false) == 0
```

Comment:
Since `datetime_eq` raises when comparing a datetime to a malformed string, it should raise when compairing with other incompatible types.

That is, you don't need `is_raising` parameter of `datetime_cmp` anymore, because it should always be raising.

Reply:
Please ask the products whether we need a compat option for this change.